sql/updates/latest-dev.sql

```diff
@@ -0,0 +1,68 @@
+-- Add dimension range tracking for a column from a hypertable.
+--
+-- hypertable - OID of the table to which the column belongs to
+-- column_name - The column to add ranges for
+-- if_not_exists - If set, and the range already exists, generate a notice instead of an error
+CREATE FUNCTION @extschema@.add_dimension_range(
+    hypertable              REGCLASS,
+    column_name             NAME,
+    if_not_exists           BOOLEAN = FALSE
+) RETURNS TABLE(dimension_range_id INT, created BOOL)
+AS '@MODULE_PATHNAME@', 'ts_column_dimension_range_add' LANGUAGE C VOLATILE;
+
+-- Remove a dimension range on a column of a hypertable.
+--
+-- hypertable - OID of the table to remove from
+-- column_name - NAME of the column on which the dimension range exists
+-- if_not_exists - If set, and the dimension range  does not exist,
+-- generate a notice instead of an error
+CREATE FUNCTION @extschema@.remove_dimension_range(
+    hypertable              REGCLASS,
+    column_name             NAME,
+    if_not_exists           BOOLEAN = FALSE
+) RETURNS TABLE(hypertable_id INT, column_name NAME, dropped BOOL)
+AS '@MODULE_PATHNAME@', 'ts_column_dimension_range_remove' LANGUAGE C VOLATILE;
+
+-- A dimension range defines a range for a given column in a chunk. It
+-- tracks the min and the max value for that column in the specific chunk.
+--
+-- A "special" entry for a column with invalid chunk_id, PG_INT64_MAX,
+-- PG_INT64_MIN indicates that ranges could be computed for this column
+-- for chunks.
+--
+-- The ranges can overlap across chunks. The values could be out-of-date if
+-- modifications/changes occur in the corresponding chunk and such entries
+-- should be marked as "invalid" to ensure that the chunk is in
+-- appropriate state to be able to use these values. Thus these entries
+-- are different from dimension_slice which is used for tracking partitioning
+-- column ranges which have different characteristics.
+--
+-- Currently this catalog supports datatypes like INT, SERIAL, BIGSERIAL,
+-- DATE, TIMESTAMP etc. by storing the ranges in bigint columns. In the
+-- future, we could support additional datatypes (which support btree style
+-- >, <, = comparators) by storing their textual representation.
+--
+CREATE TABLE _timescaledb_catalog.dimension_range (
+  id serial NOT NULL,
+  hypertable_id integer NOT NULL,
+  chunk_id integer NOT NULL,
+  column_name name NOT NULL,
+  column_type REGTYPE NOT NULL,
+  range_start bigint NOT NULL,
+  range_end bigint NOT NULL,
+  valid boolean NOT NULL,
+  -- table constraints
+  CONSTRAINT dimension_range_pkey PRIMARY KEY (id),
+  CONSTRAINT dimension_range_ht_id_chunk_id_colname_range_start_end_key UNIQUE (hypertable_id, chunk_id, column_name, range_start, range_end),
```

Comment:
Do we support multiple ranges per column? That seems odd?

If we don't, I think this constraint should be UNIQUE(hypertable_id, chunk_id, column_name); Technically, UNIQUE(chunk_id, column_name) should be enough I think.

Reply:
The idea was to avoid adding multiple entries for `ht_id, chunk_id, column_name` combo. Fair point, I will make it `UNIQUE(hypertable_id, chunk_id, column_name);`